### PR TITLE
Add JavaScriptImportScanner

### PR DESCRIPTION
**Pull request by justinfagnani**
This doesn't wire the scanner in yet, but just makes it pass a simple test.

### CODE REVIEW

File: src/javascript/javascript-import-scanner.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+// import * as estraverse from 'estraverse';
```
Comment (rictic) [inline src/javascript/javascript-import-scanner.ts]: remove?

File: src/javascript/javascript-import-scanner.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  async scan(
+      document: JavaScriptDocument,
+      visit: (visitor: Visitor) => Promise<void>): Promise<ScannedImport[]> {
+    const imports: ScannedImport[] = [];
+
+    await visit({
+      enterImportDeclaration(node: estree.ImportDeclaration, _: estree.Node) {
+        const source = node.source.value as string;
+        if (!isPathImport(source)) {
+          // TODO(justinfagnani): push a warning
```
Comment (rictic) [inline src/javascript/javascript-import-scanner.ts]: consider skipping it? otherwise we'll resolve it as though it were a relative url anyways…

Comment (justinfagnani) [inline src/javascript/javascript-import-scanner.ts]: Oh, oops, did mean to skip it. I'll add a test.

We do need a way for a scanner to report back errors/warnings though.

Comment (rictic) [inline src/javascript/javascript-import-scanner.ts]: Throw a WarningContainingException?

Comment (rictic) [inline src/javascript/javascript-import-scanner.ts]: Hm, findEntities would need to be taught about it, or maybe all of the scanners...

No, an `AddWarning` method on `Scanner` is better.

File: src/analyzer.ts [hunk 3]
```diff
@@ -57,7 +57,7 @@ export class NoKnownParserError extends Error {};
 export class Analyzer {
   private _parsers = new Map<string, Parser<ParsedDocument<any, any>>>([
     ['html', new HtmlParser()],
-    ['js', new JavaScriptParser()],
+    ['js', new JavaScriptParser({sourceType: 'script'})],
     ['css', new CssParser()],
     ['json', new JsonParser()],
```
Comment (rictic) [inline src/analyzer.ts]: You know, file extension isn't the only way we need to grab parsers.

Like, if something has been html-imported, then it needs to be handled as html. If something has been grabbed with `<script src='foo.bar>` it needs to be treated as javascript. And if something has been imported as an ES6 module it needs to be handled as a JS module.

**Comment by rictic** [comment]
This is nice to see.

LGTM once there's a test for non-relative es imports.

**Pull request merged**